public/components/common/search/autocomplete.tsx

```diff
@@ -0,0 +1,270 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+import './search.scss';
+import React, { createElement, Fragment, useEffect, useRef, useState } from 'react';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+import { RAW_QUERY } from '../../../common/constants/explorer';
+import { IQueryBarProps } from './search';
+import { handlePplRequest } from '../../../requests/ppl';
+
+// Possible suggestions (hardcoded)
+const firstCommand = [{ label: 'search' }, { label: 'source' }];
+
+const pipeCommands = [
+  { label: 'dedup' },
+  { label: 'eval' },
+  { label: 'fields' },
+  { label: 'head' },
+  { label: 'rare' },
+  { label: 'rename' },
+  { label: 'sort' },
+  { label: 'stats' },
+  { label: 'top' },
+  { label: 'where' },
+];
+
+const statsCommands = [
+  { label: 'count()' },
+  { label: 'sum()' },
+  { label: 'avg()' },
+  { label: 'max()' },
+  { label: 'min()' },
+];
+
+const indices = [
+  { label: 'opensearch_dashboards_sample_data_ecommerce' },
+  { label: 'opensearch_dashboards_sample_data_flights' },
+  { label: 'opensearch_dashboards_sample_data_logs' },
+];
+
+let inFieldsCommaLoop = false;
+let hasFields = false;
+const fieldsFromBackend = [];
+
+// Function to grab suggestions
+export function getSuggestions(str: string, http) {
+  const splittedModel = str.split(' ');
+  const prefix = splittedModel[splittedModel.length - 1];
+  let itemSuggestions = [];
+  const fullSuggestions = [];
+
+  // First commands should either be search or source
+  // source should be followed by an available index
+  if (prefix.startsWith('source=')) {
+    return getIndices(prefix.replace('source=', ''));
+  }
+  if (splittedModel.length === 1) {
+    itemSuggestions = firstCommand.filter(
+      ({ label }) => label.startsWith(prefix) && prefix !== label
+    );
+    for (let i = 0; i < itemSuggestions.length; i++) {
+      fullSuggestions.push({
+        label: str.substring(0, str.lastIndexOf(prefix)) + itemSuggestions[i].label,
+        input: str,
+        suggestion: itemSuggestions[i].label.substring(prefix.length),
+      });
+    }
+    return fullSuggestions;
+  }
+
+  if (prefix.includes(';')) {
```

Comment:
what if semicolon is part of the query? like `source = accounts employer != 'non-standard;name'`